/**********************************************************************************************************************
【文件名】                                  feedbackdlg.cpp
【功能模块和目的】                            FeedBackDlg类中成员函数的实现, 包括槽函数、接口函数
【开发者及日期】                             门恺文2019.7.26
【更改记录】
**********************************************************************************************************************/

#include "feedbackdlg.h"
#include "ui_feedbackdlg.h"
#include "confirmdlg.h"

/***********************************************************************************************************************
【函数名称】                                 FeedBackDlg()
【函数功能】                                 类FeedBackDlg的构造函数，在创建对象时被调用
【参数】                                    QWidget *parent，输入参数
【返回值】                                   无
【开发者及日期】                              门恺文 2019.7.26
【更改记录】
***********************************************************************************************************************/

FeedBackDlg::FeedBackDlg(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::FeedBackDlg)
{
    ui->setupUi(this);
}

/***********************************************************************************************************************
【函数名称】                                 ~FeedBackDlg()
【函数功能】                                 类FeedBackDlg的析构函数，在对象被销毁时调用
【参数】                                    ui
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

FeedBackDlg::~FeedBackDlg()
{
    delete ui;
}

/***********************************************************************************************************************
【函数名称】                                 setFront()
【函数功能】                                 成员front的设置函数，改变界面标题
【参数】                                     value, QString, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::setFront(const QString &value)
{
    front = value;
    ui->labFront->setText(front);
}

/***********************************************************************************************************************
【函数名称】                                 setTransId()
【函数功能】                                 成员transId的设置函数，改变界面中的提交内容
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::setTransId(int value)
{
    transId = value;
    ui->plainTextEditSubmit->setPlainText(transList[transId].getSSubmit());
}

/***********************************************************************************************************************
【函数名称】                                 setAssignId()
【函数功能】                                 成员assignId的设置函数
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::setAssignId(int value)
{
    assignId = value;
}

/***********************************************************************************************************************
【函数名称】                                 setTransCode()
【函数功能】                                 成员transCode的设置函数
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::setTransCode(int value)
{
    transCode = value;
}

/***********************************************************************************************************************
【函数名称】                                 setPncpCode()
【函数功能】                                 成员pncpCode的设置函数
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::setPncpCode(int value)
{
    pncpCode = value;
}

/***********************************************************************************************************************
【函数名称】                                 on_feedbackBtn_clicked()
【函数功能】                                 提交反馈的槽函数
【参数】                                     无
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.26
【更改记录】
***********************************************************************************************************************/

void FeedBackDlg::on_feedbackBtn_clicked()
{
    QString feedback = ui->plainTextEditFeedback->toPlainText();
    //发送前检测是否为空
    if (feedback.isEmpty())
    {
        QMessageBox::warning(this, "warning", tr("反馈意见不得为空"), QMessageBox::Ok);
        return;
    }
    //发送前再次确认
    ConfirmDlg dlg;
    QString front = tr("是否确认发送反馈意见？");
    dlg.setContent(front);
    if (dlg.exec() == QDialog::Accepted)
    {
        transList[transId].setETransState(TRANSLATING);
        transList[transId].setSFeedback(feedback);
        QMessageBox::information(this,tr("通知"), tr("反馈意见已发送"));
        //通知译者已获得反馈
        MsgBox msg(transCode, pncpCode, tr("反馈意见查收通知"), tr("您所翻译的任务")+QString::number(assignId)
                   +transList[transId].getSTitle()+tr("，负责人已给出反馈意见，请及时查看更改"));
        msgList.push_back(msg);
    }
}
